### PR TITLE
feat: add temporary indexer and data source

### PR DESCRIPTION
**Pull request by jrabbott**
### Context
[AB#210705](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/210705)

### Change proposed in this pull request
Temporarily add indexers and data sources to seed school comparator index

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

File: platform/src/search/Platform.Search/SchoolComparators/SchoolComparatorsAcademyIndexerBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public override async Task Build(SearchIndexerClient client)
+    {
+        var cosmosDbIndexer = new SearchIndexer(
+            name: Name,
+            dataSourceName: SearchResourceNames.DataSources.SchoolComparatorsAcademy,
+            targetIndexName: SearchResourceNames.Indexes.SchoolComparators)
+        {
+            FieldMappings =
+            {
+                new FieldMapping("URN") { TargetFieldName = nameof(SchoolComparatorsIndex.UKPRN) },
```
Comment (WolfyUK) [inline platform/src/search/Platform.Search/SchoolComparators/SchoolComparatorsAcademyIndexerBuilder.cs:21]: Just checking this particular mapping is intentional

File: platform/src/search/Platform.Search/SchoolComparators/SchoolComparatorsMaintainedIndexerBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public override async Task Build(SearchIndexerClient client)
+    {
+        var cosmosDbIndexer = new SearchIndexer(
+            name: Name,
+            dataSourceName: SearchResourceNames.DataSources.SchoolComparatorsAcademy,
+            targetIndexName: SearchResourceNames.Indexes.SchoolComparators)
+        {
+            FieldMappings =
+            {
+                new FieldMapping("URN") { TargetFieldName = nameof(SchoolComparatorsIndex.UKPRN) },
```
Comment (WolfyUK) [inline platform/src/search/Platform.Search/SchoolComparators/SchoolComparatorsMaintainedIndexerBuilder.cs:21]: ...and here

File: platform/src/search/Platform.Search/SchoolComparators/SchoolComparatorsAcademyIndexerBuilder.cs (hunk 1 above)
Comment (jrabbott) [reply platform/src/search/Platform.Search/SchoolComparators/SchoolComparatorsAcademyIndexerBuilder.cs:21]: Yeah, just a hack at the moment to get some data in the index to tweak the analyzers and lucene syntax

File: platform/src/search/Platform.Search/SchoolComparators/SchoolComparatorsMaintainedIndexerBuilder.cs (hunk 2 above)
Comment (jrabbott) [reply platform/src/search/Platform.Search/SchoolComparators/SchoolComparatorsMaintainedIndexerBuilder.cs:21]: Same

**Review by WolfyUK — APPROVED** [review]

**Pull request merged**